fix compression of empty string

esc opened the pull request:
The check here was faulty, in case of the empty string, the output
buffer will be exactly BLOSC_MAX_OVERHEAD. I.e. the output buffer will
contain only exactly the blosc header and nothing else.

esc commented:
`master` is currently failing it would seem:

```
     Start 669: test_maxout
669/1640 Test #669: test_maxout ..................................***Failed    0.00 sec
```

danstender commented:
On 11/20/18 12:50 PM, Emergency Self-Construct wrote:
> |master| is currently failing it would seem:
> 
> |Start 669: test_maxout 669/1640 Test #669: test_maxout .........***Failed 0.00 sec |

Ah, o.k. Thanks! ... but that one run through here [1]:

```
         Start 668: test_maxout
665/1639 Test #668: test_maxout ..................................   Passed    0.00 sec
```

What's going on?

[1] https://buildd.debian.org/status/fetch.php?pkg=c-blosc&arch=i386&ver=1.14.4

FrancescAlted commented:
Yes, that was an oversight on my part.  Could you try to fix the `test_maxout` issue?

esc commented:
@danstender I can't find the build log for debian, has it expired?

esc commented:
Anyway, rebasing onto master

esc commented:
O.K. I can confirm that this affects only the branch and that I can reproduce this locally on my machine.

esc commented:
O.K. found the error and added a commit, let's see what Travis says this time.

kalvdans commented:
Any idea why the test hanged instead of failed?

danstender commented:
@esc the logs of our build of 1.14.4 are here: https://buildd.debian.org/status/package.php?p=c-blosc

esc commented:
@danstender thanks, the error only happend in the branch for this PR, it is fixed now.

esc commented:
@danstender I added a test for the empty buffer and rebased the commits such that the test to expose the bug comes BEFORE the fix. This should help to verify that the fix does actually fix the bug. Let's wait for CI to finish.

esc commented:
Hmmm, it seems like github isn't picking up on the interactive rebase I used to re-order the commits, haha.

kalvdans commented:
@esc I meant https://github.com/Blosc/c-blosc/issues/246, so I asked there instead. Sorry for adding noise here.

danstender commented:
... I can rebuild Git snapshots anytime in experimental, just tell if you would like to see that.

esc commented:
@danstender : 
"... I can rebuild Git snapshots anytime in experimental, just tell if you would like to see that."

What does the above mean, I know a little about Debian packaging, but I don't understand the above comment, can you elaborate a little, please.

esc commented:
@kalvdans no worries, I responded over on the other issue ;)

esc commented:
Hmmm, it really does seem like Github/Travis got kinda confused by my interactive rebase concoction, looks like it will be a while until the tests complete.

danstender commented:
... meaning that I could make a package for the Debian experimental branch which isn't for release but only for testing issues. But on the other hand, if it doesn't build because a test failure there wouldn't be any package to upload. It's useful mostly if it builds on amd64, but wanted to try out it does on the other ports/archs. But there's Travis, that's allright ...

esc commented:
@FrancescAlted @danstender so, Travis is green, which is good. The bug has been fixed, the tests have been fixed and I included a test for compression of the empty buffer to ensure that we can catch any regressions of this issue.

I would propose to merge this pull-request and release 1.14.2 - because I will then update the vendored copy in python-blosc and release a new version. Also, I think Dan would like to build a new Debian package too, right?

esc commented:
Cool, timeline for the buster freeze is here:

https://release.debian.org/buster/freeze_policy.html

FrancescAlted commented:
Cool, let me see if I can do a release before the freeze :-P

esc commented:
@FrancescAlted awesome! Thanks for merging!

FrancescAlted commented:
Release 1.15.0 is out.  Enjoy!
